### ISSUE TITLE
RFC: New 6px baseline grid to enable 16px / 24px standard text

### ISSUE THREAD

**Issue by markdalgleish** [seek-oss/seek-style-guide#312]
# Summary

Migrate from a 9px grid to a 6px grid to allow for larger 16px body copy over a 24px line height across the site.

# Motivation

Our current `standard` text size is 14px / 18px on desktop and 18px / 27px on mobile. We also have `substandard` and `superstandard` variants which, respectively, maintain 14px / 18px or 18px / 27px across both desktop and mobile. In many cases, both of these are causing us issues:

- For dense, skimmable content (e.g. search results), 14px is smaller than we would like on mobile, but 18px is much too big. We'd choose a font size between these two, but it doesn't sit correctly on our current 9px grid, generating excess amounts of leading.
- For more long-form content on desktop, 18px seems to be more preferable, but our heading sizes were not designed with 18px body copy in mind. When using `superstandard` text in combination with the full range of headings, it can lead to very unbalanced typography.

Another issue to consider is that, in practice, the 9px grid often creates too much whitespace by forcing elements to either be too close or too far from each other. If we can improve the amount of design flexibility, without undoing all the benefits of having a baseline grid, that would be a positive outcome too.

# Detailed design

If we want to properly solve this issue, we find ourselves naturally gravitating towards a pretty substantial change—discarding the current 9px grid, settling on a larger body copy size that works *everywhere* (without any sub/super variant hacks), and using this to form the basis a new baseline grid.

From a design perspective, we quickly landed on 16px / 24px, which leaves us with a grid system based on either 8px, 6px or 4px. After much experimentation across a wide range of projects, we landed on 6px as it provided the best balance between project-level flexibility and system-level rigidity.

Since the grid variables are used in many calculations throughout the site, changing their values will no doubt cause widespread breakage. As a result, I think we should introduce entirely new grid variables so that consumers have a chance to upgrade without being forced to convert every single grid-based calculation at once, e.g.

```less
@grid-row: 6px; // New
@grid-row-height: 9px; // Old
 ```

Our horizontal grid values shouldn't need to change, but, for consistency, we might introduce new names for these too, e.g.

```less
@grid-gutter: 20px; // New
@grid-gutter-width: 20px; // Old
 ```

So far, the new type hierarchy based on a 6px baseline grid looks like this:

Desktop:
- Substandard/Superstandard: DEPRECATED 🎉
- Standard: 16px / 4 (24px)
- Small: 14px / 3 (18px)
- Interaction: 18px / 8 (48px)
- Subheading: 18px / 4 (24px)
- Heading: 21px / 5 (30px)
- Headline: 28px / 6 (36px)
- Hero: 42px / 8 (48px)

Mobile:
- Substandard/Superstandard: DEPRECATED 🎉🎉
- Standard: 16px / 4 (24px)
- Small: 14px / 3 (18px)
- Interaction: 18px / 8 (48px)
- Subheading: 18px / 4 (24px)
- Heading: 21px / 5 (30px)
- Headline: 21px / 5 (30px)
- Hero: 21px / 5 (30px)

Components that make use of these variables internally will need to be migrated, but in theory they should be able to be upgraded without too much trouble.

# How We Teach This

We'll refer to the new variables going forwards and remove references to the old ones, but otherwise the style guide remains largely the same.

# Drawbacks

Any project that uses both new and old grid variables at the same time will no longer be on-grid. This isn't a huge problem, assuming we fully migrate away from the old variables sooner rather than later.

# Alternatives

We could opt for an 8px grid instead, although this doesn't give us much more flexility than we have currently.

We could also leave the current variable names in place and simply change their values, but as discussed earlier, this would drastically increase the odds of breaking existing layouts.

# Unresolved questions

How do we handle the shared header, which is on the 9px grid but is used on pages that aren't yet using the style guide? Should we update the header to follow the new grid, or leave it alone? Is it okay if the header changes height slightly between newer and older pages?

Should we change the horizontal grid variables to match the 6px grid? Now would be a good time to do it, if we wanted to.

**Comment by mattsjones** [comment]
As this would be such a large change for each app requiring lots of small tweaks to the design how do we plan to roll this out? The main problem I see is with the `Text` component. 

I think we might need to create a `legacy` prop or something which would switch between the new grid system and the old. Otherwise other style guide upgrades will be blocked by this.

**Comment by markdalgleish** [comment]
Not that I disagree, but what are your specific concerns with the `Text` component?

**Comment by markdalgleish** [comment]
@mattsjones ping 😄

**Comment by mattsjones** [comment]
Sorry missed your reply. My main concern is if you change the value of `grid-row-height` then that will affect all the internal style guide components that use it. How do we plan on giving people a way of migrating slowly to the new grid system for the style guide components? `Text` is just the widest spread one that will be affected by this.

**Comment by markdalgleish** [comment]
My thinking was that the components would just use the new grid system without a way to opt out—not sure it's worth the hassle. My concern is more around changing the value of `@grid-row-height` which would cause a lot of breakages, which is why I suggested we introduce new variables instead and leave the existing variables for backwards compatibility.

**Comment by mengtzu** [comment]
This seems fine.  My only concern is getting off the old variables as quickly as possible, they will be SUPER confusing in context if you didn't know about this change.
